---
title: "GWAS"
knit: (function(input_file, encoding) {
  out_dir <- 'docs';
  rmarkdown::render(input_file,
 encoding=encoding,
 output_file=file.path(dirname(input_file), out_dir, 'GWAS.html'))})
 
author: "Ying Wang"
date: "15/01/2021"
output: 
  html_document:
    code_folding: hide
    toc: true
    toc_float: true
    number_sections: TRUE
---
  
```{r setup, include=FALSE}
WD="~/OneDrive - The University of Queensland/mater_projs/auti_proj/AutismSpeaks_SSC/"
knitr::opts_knit$set(root.dir = WD)

knitr::opts_chunk$set(echo = TRUE)
knitr::opts_chunk$set(fig.pos = 't')
library(DT)
library(png)
library(grid)
```

***
  
  <br>
  
# GWAS Strategy
GWAS was run using MLM model in [GCTA1.93.2](https://cnsgenomics.com/software/gcta/). Note that I tried different strategies to directly fit covariates or pre-adjust phenotypes by the covariates. The beta correlation between different strategies will be shown as below. 

Take SRS_RMB_sum in Probands (with FSIQ included as covariate) for example, I tried: 
  
- Strategy 1: Phenotype pre-adjusted by age, sex, chip, FSIQ
- Strategy 2: Phenotype pre-adjusted by age, sex, chip, FSIQ and 20 PCs
- Strategy 3: directly fit age, sex, chip, FSIQ
- Strategy 4: directly fit age, sex, chip, FSIQ and 20 PCs

```{r gwas_Phenotype, eval=FALSE, include=FALSE}
setwd("/scratch/90days/uqywan67/auti_proj/SSC/GWAS")
phename <- "SRS_RMB_sum"
gwas1 <- read.table(paste0("ssc_probands_adjIQ_noPCs_", phename, ".mlma"), header = T, stringsAsFactors = F)
gwas2 <- read.table(paste0("ssc_probands_adjIQ_withPCs_", phename, ".mlma"), header = T, stringsAsFactors = F)
gwas3 <- read.table(paste0("ssc_probands_adjIQ2_noPCs_", phename, ".mlma"), header = T, stringsAsFactors = F)
gwas4 <- read.table(paste0("ssc_probands_adjIQ2_withPCs_", phename, ".mlma"), header = T, stringsAsFactors = F)

library(ggpmisc)
library(reshape2)
res <- data.frame(gwas1$b, gwas2$b, gwas3$b, gwas4$b)
names(res) <- paste0("Strategy", 1:4)
res1 <- melt(res, id = "Strategy1")

p_gwas <- ggplot(data = res1, aes(y = value, x = Phenotype1)) +
  geom_smooth(method = "lm", se=TRUE, formula = y~x) +
  stat_poly_eq(formula = y~x, 
               aes(label = paste(..eq.label.., ..rr.label.., sep = "~~~")), 
               parse = TRUE,label.x.npc = "left", label.y.npc = "top",vjust=0.8) +         
  geom_point()+
  facet_wrap(~variable, nrow = 1)
theme_bw()+
  labs(x="Beta in Strategy1",y="Beta in another Strategy")+
  #ggtitle("Beta correlation between UKB2-GWS VS Test sets")+
  theme(
    plot.title = element_text(hjust = 0.5),
    legend.position="none"
  )

ggsave(file.path("beta_strategy.png"), p_gwas, height = 6, width = 14, dpi = 300)

```

```{r plot_gwas, out.height="80%", out.width="100%", fig.cap="Beta Correlation between Different GWAS Strategies"}
#grid.raster(readPNG("figures/beta_strategy.png")
grid.raster(readPNG("figures/beta_strategy.png"))
```

<br>
  
**Note** that considering the GWAS sample size, computational time and false positive rates,  we will report the results below:
  
- based on **Strategy2** for the same phenotype
- based on **sum** measurement for the same phenotype

```{r merge_GWAS_probands, eval=FALSE, include=FALSE}
setwd("/scratch/90days/uqywan67/auti_proj/SSC/outputs")
phe_unaff <- read.table("/afm01/UQ/Q3490/SSC/phenotypes/Unaffected_Siblings_matchedID_cleaned.txt", header = T, stringsAsFactors = F)
phes_unaff <- colnames(phe_unaff)

summs <- read.table("phenotype_variable_summary.txt", header = T, stringsAsFactors = F)

iqs <- read.table("summary_probands_p1e5.txt", stringsAsFactors = F)
noiqs <- read.table("summary_probands_p1e5_noIQ.txt", stringsAsFactors = F)
names(iqs) <- names(noiqs) <- c("Chr", "SNP", "bp", "A1", "A2", "Freq", "b", "se", "p", "Strategy", "Phenotype")

iqs$Interests <- "Primary"
iqs[iqs$Phenotype %in% summs$Primary_Variable, "Interests"] <- "Primary"
iqs[iqs$Phenotype %in% summs$Secondary_Variables, "Interests"] <- "Secondary"
iqs$Shared <- "No"
iqs[iqs$Phenotype %in% phes_unaff, "Shared"] <- "Yes"

iqs1 <- iqs[grep("sum", iqs$Phenotype),]
iqs2 <- iqs1[iqs1$Strategy == "Strategy2",][,c(1:3, 7:13)]
names(iqs2)[1:6] <- c("CHR", "SNP", "BP", "b", "se", "P")

write.table(iqs2, file = "summary_probands_p1e5_strategy2.txt", col.names = T, row.names = F, quote = F, sep = "\t")
write.table(iqs, file = "summary_probands_p1e5.txt", col.names = T, row.names = F, quote = F, sep = "\t")

noiqs$Interests <- "Primary"
noiqs[noiqs$Phenotype %in% summs$Primary_Variable, "Interests"] <- "Primary"
noiqs[noiqs$Phenotype %in% summs$Secondary_Variables, "Interests"] <- "Secondary"
noiqs$Shared <- "No"
noiqs[noiqs$Phenotype %in% phes_unaff, "Shared"] <- "Yes"

noiqs1 <- noiqs[grep("sum", noiqs$Phenotype),]
noiqs2 <- noiqs1[noiqs1$Strategy == "Strategy2",][,c(1:3, 7:13)]
names(noiqs2)[1:6] <- c("CHR", "SNP", "BP", "b", "se", "P")
#noiqs2$zscore <- noiqs2$b/noiqs2$se
write.table(noiqs2, file = "summary_probands_p1e5_noIQ_strategy2.txt", col.names = T, row.names = F, quote = F, sep = "\t")
write.table(noiqs, file = "summary_probands_p1e5_noIQ.txt", col.names = T, row.names = F, quote = F, sep = "\t")

```

```{r include=FALSE}
iqs2 <- read.table("../outputs/summary_probands_p1e5_strategy2.txt", header = T, stringsAsFactors = F)
iqs2_1 <- iqs2[iqs2$Interests == "Primary",]
iqs2_2 <- iqs2[iqs2$Interests == "Secondary",]

noiqs2 <- read.table("../outputs/summary_probands_p1e5_noIQ_strategy2.txt", header = T, stringsAsFactors = F)
noiqs2_1 <- noiqs2[noiqs2$Interests == "Primary",]
noiqs2_2 <- noiqs2[noiqs2$Interests == "Secondary",]

```

```{r plot_manha, eval=FALSE, include=FALSE}
##with IQ adjusted
png("figures/manhplot_probands_adjIQ_1e-5_primary_withPCs.png",width=35, height = 21, units = "in", res = 300)
par(mfrow=c(3,5))
par(mar=c(5,5,5,5))
for(phename in levels(factor(iqs2_1$Phenotype))){
  dat <- iqs2_1[iqs2_1$Phenotype == phename,]
  phename1 <- gsub("_sum", "", phename)
  manhattan(dat, main = phename1, ylim = c(5, 10), col = c("blue4", "orange3"), suggestiveline = F, genomewideline = -log10(5e-08), cex = 3, annotatePval = 5e-8, annotateTop = F, cex.axis = 0.8, cex.main = 3, cex.lab =2, cex.axis = 2
  )
}
dev.off()

png("figures/manhplot_probands_adjIQ_1e-5_secondary_withPCs.png",width=28, height = 21, units = "in", res = 300)
par(mfrow=c(3,4))
par(mar=c(5,5,5,5))
for(phename in levels(factor(iqs2_2$Phenotype))){
  dat <- iqs2_2[iqs2_2$Phenotype == phename,]
  phename1 <- gsub("_sum", "", phename)
  manhattan(dat, main = phename1, ylim = c(5, 10), col = c("blue4", "orange3"), suggestiveline = F, genomewideline = -log10(5e-08), cex = 3, annotatePval = 5e-8, annotateTop = F, cex.axis = 0.8, cex.main = 3, cex.lab =2, cex.axis = 2
  )
}
dev.off()

##without IQ adjusted
png("figures/manhplot_probands_noIQ_1e-5_primary_withPCs.png",width=35, height = 21, units = "in", res = 300)
par(mfrow=c(3,5))
par(mar=c(5,5,5,5))
for(phename in levels(factor(noiqs2_1$Phenotype))){
  dat <- noiqs2_1[noiqs2_1$Phenotype == phename,]
  phename1 <- gsub("_sum", "", phename)
  manhattan(dat, main = phename1, ylim = c(5, 10), col = c("blue4", "orange3"), suggestiveline = F, genomewideline = -log10(5e-08), cex = 3, annotatePval = 5e-8, annotateTop = F, cex.axis = 0.8, cex.main = 3, cex.lab =2, cex.axis = 2
  )
}
dev.off()

png("figures/manhplot_probands_noIQ_1e-5_secondary_withPCs.png",width=28, height = 21, units = "in", res = 300)
par(mfrow=c(3,4))
par(mar=c(5,5,5,5))
for(phename in levels(factor(noiqs2_2$Phenotype))){
  dat <- noiqs2_2[noiqs2_2$Phenotype == phename,]
  phename1 <- gsub("_sum", "", phename)
  manhattan(dat, main = phename1, ylim = c(5, 10), col = c("blue4", "orange3"), suggestiveline = F, genomewideline = -log10(5e-08), cex = 3, annotatePval = 5e-8, annotateTop = F, cex.axis = 0.8, cex.main = 3, cex.lab =2, cex.axis = 2
  )
}
dev.off()
```


***
  
<br> 
  
# Probands  
## All Individuals  {.tabset .tabset-fade .tabset-pills}
- GWAS was run on all individuals including diverse ancestry backgrounds.
- Signals with association p-value < 1e-5 will be shown.

### Association Summary {.tabset .tabset-fade .tabset-pills}
#### Fitting FSIQ
```{r}
datatable(iqs2, rownames = FALSE, filter="top", options = list(pageLength = 5, scrollX=T) )
```

***
  
<br>

#### Not fitting FSIQ
```{r}
datatable(noiqs2, rownames = FALSE, filter="top", options = list(pageLength = 5, scrollX=T) )
```

***

<br>

### Manhattan Plot {.tabset .tabset-fade .tabset-pills}
#### Primary Variable  {.tabset .tabset-fade .tabset-pills}
##### Fitting FSIQ
```{r out.width="100%", out.height="100%"}
#grid.raster(readPNG("figures/manhplot_probands_adjIQ_1e-5_primary_withPCs.png")
img <- readPNG("figures/manhplot_probands_adjIQ_1e-5_primary_withPCs.png")
grid.raster(img)
```

***
  
<br>

##### Not fitting FSIQ
```{r out.width="100%", out.height="100%"}
grid.raster(readPNG("figures/manhplot_probands_noIQ_1e-5_primary_withPCs.png"))
```


***
  
<br>

#### Secondary Variable  {.tabset .tabset-fade .tabset-pills}

##### Fitting FSIQ
```{r out.width="100%", out.height="100%"}
grid.raster(readPNG("figures/manhplot_probands_adjIQ_1e-5_secondary_withPCs.png"))
```

***
  
<br>

##### Not fitting FSIQ
```{r out.width="100%", out.height="100%"}
grid.raster(readPNG("figures/manhplot_probands_noIQ_1e-5_secondary_withPCs.png"))
```

***
  
<br>

## Europeans Only  {.tabset .tabset-fade .tabset-pills}
- 6861972 QCd SNPs with MAF > 0.01 included
- 1946 European individuals are included
- Signals with association p-value < 1e-5 will be shown.
```{r merge_GWAS_probands_EUR, eval=FALSE, include=FALSE}
setwd("/scratch/90days/uqywan67/auti_proj/SSC/outputs")
phe_unaff <- read.table("../phenotypes/Unaffected_Siblings_matchedID_cleaned.txt", header = T, stringsAsFactors = F)
phes_unaff <- colnames(phe_unaff)

summs <- read.table("phenotype_variable_summary.txt", header = T, stringsAsFactors = F)

iqs_EUR <- read.table("summary_probands_p1e5_EUR.txt", stringsAsFactors = F)
noiqs_EUR <- read.table("summary_probands_p1e5_noIQ_EUR.txt", stringsAsFactors = F)
names(iqs_EUR) <- names(noiqs_EUR) <- c("Chr", "SNP", "bp", "A1", "A2", "Freq", "b", "se", "p", "Strategy", "Phenotype")

iqs_EUR$Interests <- "Primary"
iqs_EUR[iqs_EUR$Phenotype %in% summs$Primary_Variable, "Interests"] <- "Primary"
iqs_EUR[iqs_EUR$Phenotype %in% summs$Secondary_Variables, "Interests"] <- "Secondary"
iqs_EUR$Shared <- "No"
iqs_EUR[iqs_EUR$Phenotype %in% phes_unaff, "Shared"] <- "Yes"

iqs_EUR1 <- iqs_EUR[grep("sum", iqs_EUR$Phenotype),]
iqs_EUR2 <- iqs_EUR1[iqs_EUR1$Strategy == "Strategy2",][,c(1:3, 7:13)]
names(iqs_EUR2)[1:6] <- c("CHR", "SNP", "BP", "b", "se", "P")

write.table(iqs_EUR2, file = "summary_probands_p1e5_strategy2_EUR.txt", col.names = T, row.names = F, quote = F, sep = "\t")
write.table(iqs_EUR, file = "summary_probands_p1e5_EUR.txt", col.names = T, row.names = F, quote = F, sep = "\t")

noiqs_EUR$Interests <- "Primary"
noiqs_EUR[noiqs_EUR$Phenotype %in% summs$Primary_Variable, "Interests"] <- "Primary"
noiqs_EUR[noiqs_EUR$Phenotype %in% summs$Secondary_Variables, "Interests"] <- "Secondary"
noiqs_EUR$Shared <- "No"
noiqs_EUR[noiqs_EUR$Phenotype %in% phes_unaff, "Shared"] <- "Yes"

noiqs_EUR1 <- noiqs_EUR[grep("sum", noiqs_EUR$Phenotype),]
noiqs_EUR2 <- noiqs_EUR1[noiqs_EUR1$Strategy == "Strategy2",][,c(1:3, 7:13)]
names(noiqs_EUR2)[1:6] <- c("CHR", "SNP", "BP", "b", "se", "P")
#noiqs_EUR2$zscore <- noiqs_EUR2$b/noiqs_EUR2$se
write.table(noiqs_EUR2, file = "summary_probands_p1e5_noIQ_strategy2_EUR.txt", col.names = T, row.names = F, quote = F, sep = "\t")
write.table(noiqs_EUR, file = "summary_probands_p1e5_noIQ_EUR.txt", col.names = T, row.names = F, quote = F, sep = "\t")

```

```{r load_var, include=FALSE}
iqs_EUR2 <- read.table("../outputs/summary_probands_p1e5_strategy2_EUR.txt", header = T, stringsAsFactors = F)
iqs_EUR2_1 <- iqs_EUR2[iqs_EUR2$Interests == "Primary",]
iqs_EUR2_2 <- iqs_EUR2[iqs_EUR2$Interests == "Secondary",]

noiqs_EUR2 <- read.table("../outputs/summary_probands_p1e5_noIQ_strategy2_EUR.txt", header = T, stringsAsFactors = F)
noiqs_EUR2_1 <- noiqs_EUR2[noiqs_EUR2$Interests == "Primary",]
noiqs_EUR2_2 <- noiqs_EUR2[noiqs_EUR2$Interests == "Secondary",]
```

```{r plot_manha_EUR, eval=FALSE, include=FALSE}
##with IQ adjusted
png("figures/manhplot_probands_adjIQ_1e-5_primary_EUR_withPCs.png",width=35, height = 21, units = "in", res = 300)
par(mfrow=c(3,5))
par(mar=c(5,5,5,5))
for(phename in levels(factor(iqs_EUR2_1$Phenotype))){
  dat <- iqs_EUR2_1[iqs_EUR2_1$Phenotype == phename,]
  phename1 <- gsub("_sum", "", phename)
  manhattan(dat, main = phename1, ylim = c(5, 10), col = c("blue4", "orange3"), suggestiveline = F, genomewideline = -log10(5e-08), cex = 3, annotatePval = 5e-8, annotateTop = F, cex.axis = 0.8, cex.main = 3, cex.lab =2, cex.axis = 2
  )
}
dev.off()

png("figures/manhplot_probands_adjIQ_1e-5_secondary_EUR_withPCs.png",width=28, height = 21, units = "in", res = 300)
par(mfrow=c(3,4))
par(mar=c(5,5,5,5))
for(phename in levels(factor(iqs_EUR2_2$Phenotype))){
  dat <- iqs_EUR2_2[iqs_EUR2_2$Phenotype == phename,]
  phename1 <- gsub("_sum", "", phename)
  manhattan(dat, main = phename1, ylim = c(5, 10), col = c("blue4", "orange3"), suggestiveline = F, genomewideline = -log10(5e-08), cex = 3, annotatePval = 5e-8, annotateTop = F, cex.axis = 0.8, cex.main = 3, cex.lab =2, cex.axis = 2
  )
}
dev.off()

##without IQ adjusted
png("figures/manhplot_probands_noIQ_1e-5_primary_EUR_withPCs.png",width=35, height = 21, units = "in", res = 300)
par(mfrow=c(3,5))
par(mar=c(5,5,5,5))
for(phename in levels(factor(noiqs_EUR2_1$Phenotype))){
  dat <- noiqs_EUR2_1[noiqs_EUR2_1$Phenotype == phename,]
  phename1 <- gsub("_sum", "", phename)
  manhattan(dat, main = phename1, ylim = c(5, 10), col = c("blue4", "orange3"), suggestiveline = F, genomewideline = -log10(5e-08), cex = 3, annotatePval = 5e-8, annotateTop = F, cex.axis = 0.8, cex.main = 3, cex.lab =2, cex.axis = 2
  )
}
dev.off()

png("figures/manhplot_probands_noIQ_1e-5_secondary_EUR_withPCs.png",width=28, height = 21, units = "in", res = 300)
par(mfrow=c(3,4))
par(mar=c(5,5,5,5))
for(phename in levels(factor(noiqs_EUR2_2$Phenotype))){
  dat <- noiqs_EUR2_2[noiqs_EUR2_2$Phenotype == phename,]
  phename1 <- gsub("_sum", "", phename)
  manhattan(dat, main = phename1, ylim = c(5, 10), col = c("blue4", "orange3"), suggestiveline = F, genomewideline = -log10(5e-08), cex = 3, annotatePval = 5e-8, annotateTop = F, cex.axis = 0.8, cex.main = 3, cex.lab =2, cex.axis = 2
  )
}
dev.off()
```


***
  
<br> 

### Association Summary   {.tabset .tabset-fade .tabset-pills}
#### Fitting FSIQ
```{r}
datatable(iqs_EUR2, rownames = FALSE, filter="top", options = list(pageLength = 5, scrollX=T) )
```

***
  
<br>

#### Not fitting FSIQ
```{r}
datatable(noiqs_EUR2, rownames = FALSE, filter="top", options = list(pageLength = 5, scrollX=T) )
```

***
  
<br>

### Manhattan Plot {.tabset .tabset-fade .tabset-pills}
#### Primary Variable  {.tabset .tabset-fade .tabset-pills}
##### Fitting FSIQ
```{r out.width="100%", out.height="100%"}
grid.raster(readPNG("figures/manhplot_probands_adjIQ_1e-5_primary_EUR_withPCs.png"))
```

***
  
<br>

##### Not fitting FSIQ
```{r out.width="100%", out.height="100%"}
grid.raster(readPNG("figures/manhplot_probands_noIQ_1e-5_primary_EUR_withPCs.png"))
```


***
  
<br>

#### Secondary Variable  {.tabset .tabset-fade .tabset-pills}

##### Fitting FSIQ
```{r out.width="100%", out.height="100%"}
grid.raster(readPNG("figures/manhplot_probands_adjIQ_1e-5_secondary_EUR_withPCs.png"))
```

***
  
<br>

##### Not fitting FSIQ
```{r out.width="100%", out.height="100%"}
grid.raster(readPNG("figures/manhplot_probands_noIQ_1e-5_secondary_EUR_withPCs.png"))
```

***
  
<br>

# Probands & Unaffected Siblings  
- Phenotypes for Probands and Unaffected Siblings are separately pre-adjusted by covariates and then RINT.
- Based on the phenotype distribution, we combine Probands and Unaffected Siblings.
- We run GWAS on combined data.

## All Individuals {.tabset .tabset-fade .tabset-pills}
- GWAS was run on all individuals including diverse ancestry backgrounds.
- Signals with association p-value < 1e-5 will be shown.

```{r merge_GWAS_probandsSibs, eval=FALSE, include=FALSE}
setwd("/scratch/90days/uqywan67/auti_proj/SSC/outputs")
phe_unaff <- read.table("../Unaffected_Siblings_matchedID_cleaned.txt", header = T, stringsAsFactors = F)
phes_unaff <- colnames(phe_unaff)

summs <- read.table("phenotype_variable_summary.txt", header = T, stringsAsFactors = F)

iqs_probSibs <- read.table("summary_probandsAndunaffSibs_p1e5.txt", stringsAsFactors = F)
noiqs_probSibs <- read.table("summary_probandsAndunaffSibs_p1e5_noIQ.txt", stringsAsFactors = F)
names(iqs_probSibs) <- names(noiqs_probSibs) <- c("Chr", "SNP", "bp", "A1", "A2", "Freq", "b", "se", "p", "Strategy", "Phenotype")

iqs_probSibs$Interests <- "Primary"
iqs_probSibs[iqs_probSibs$Phenotype %in% summs$Primary_Variable, "Interests"] <- "Primary"
iqs_probSibs[iqs_probSibs$Phenotype %in% summs$Secondary_Variables, "Interests"] <- "Secondary"

iqs_probSibs1 <- iqs_probSibs[grep("sum", iqs_probSibs$Phenotype),]
iqs_probSibs2 <- iqs_probSibs1[iqs_probSibs1$Strategy == "Strategy2",][,c(1:3, 6:12)]
names(iqs_probSibs2)[1:7] <- c("CHR", "SNP", "BP", "Freq",  "b", "se", "P")

write.table(iqs_probSibs2, file = "summary_probandsAndUnaffSibs_p1e5_strategy2.txt", col.names = T, row.names = F, quote = F, sep = "\t")
write.table(iqs_probSibs, file = "summary_probandsAndUnaffSibs_p1e5.txt", col.names = T, row.names = F, quote = F, sep = "\t")

noiqs_probSibs$Interests <- "Primary"
noiqs_probSibs[noiqs_probSibs$Phenotype %in% summs$Primary_Variable, "Interests"] <- "Primary"
noiqs_probSibs[noiqs_probSibs$Phenotype %in% summs$Secondary_Variables, "Interests"] <- "Secondary"


noiqs_probSibs1 <- noiqs_probSibs[grep("sum", noiqs_probSibs$Phenotype),]
noiqs_probSibs2 <- noiqs_probSibs1[noiqs_probSibs1$Strategy == "Strategy2",][,c(1:3, 6:12)]
names(noiqs_probSibs2)[1:7] <- c("CHR", "SNP", "BP", "Freq", "b", "se", "P")
#noiqs_probSibs2$zscore <- noiqs_probSibs2$b/noiqs_probSibs2$se
write.table(noiqs_probSibs2, file = "summary_probandsAndUnaffSibs_p1e5_noIQ_strategy2.txt", col.names = T, row.names = F, quote = F, sep = "\t")
write.table(noiqs_probSibs, file = "summary_probandsAndUnaffSibs_p1e5_noIQ.txt", col.names = T, row.names = F, quote = F, sep = "\t")

```



```{r include=FALSE}
iqs_probSibs2 <- read.table("../outputs/summary_probandsAndUnaffSibs_p1e5_strategy2.txt", header = T, stringsAsFactors = F)
iqs_probSibs2_1 <- iqs_probSibs2[iqs_probSibs2$Interests == "Primary",]
iqs_probSibs2_2 <- iqs_probSibs2[iqs_probSibs2$Interests == "Secondary",]

noiqs_probSibs2 <- read.table("../outputs/summary_probandsAndUnaffSibs_p1e5_noIQ_strategy2.txt", header = T, stringsAsFactors = F)
noiqs_probSibs2_1 <- noiqs_probSibs2[noiqs_probSibs2$Interests == "Primary",]
noiqs_probSibs2_2 <- noiqs_probSibs2[noiqs_probSibs2$Interests == "Secondary",]

```

```{r plot_manha_probSibs, eval=FALSE, include=FALSE}
##with IQ adjusted
png("figures/manhplot_probandsAndUnaffSibs_adjIQ_1e-5_primary_withPCs.png",width=12, height = 12, units = "in", res = 300)
par(mfrow=c(2,2))
par(mar=c(5,5,5,5))
for(phename in levels(factor(iqs_probSibs2_1$Phenotype))){
  dat <- iqs_probSibs2_1[iqs_probSibs2_1$Phenotype == phename,]
  phename1 <- gsub("_sum", "", phename)
  manhattan(dat, main = phename1, ylim = c(5, 10), col = c("blue4", "orange3"), suggestiveline = F, genomewideline = -log10(5e-08), cex = 3, annotatePval = 5e-8, annotateTop = F, cex.axis = 0.8, cex.main = 3, cex.lab =2, cex.axis = 2
  )
}
dev.off()

png("figures/manhplot_probandsAndUnaffSibs_adjIQ_1e-5_secondary_withPCs.png",width=18, height = 6, units = "in", res = 300)
par(mfrow=c(1,3))
par(mar=c(5,5,5,5))
for(phename in levels(factor(iqs_probSibs2_2$Phenotype))){
  dat <- iqs_probSibs2_2[iqs_probSibs2_2$Phenotype == phename,]
  phename1 <- gsub("_sum", "", phename)
  manhattan(dat, main = phename1, ylim = c(5, 10), col = c("blue4", "orange3"), suggestiveline = F, genomewideline = -log10(5e-08), cex = 3, annotatePval = 5e-8, annotateTop = F, cex.axis = 0.8, cex.main = 3, cex.lab =2, cex.axis = 2
  )
}
dev.off()

##without IQ adjusted
png("figures/manhplot_probandsAndUnaffSibs_noIQ_1e-5_primary_withPCs.png",width=12, height = 12, units = "in", res = 300)
par(mfrow=c(2,2))
par(mar=c(5,5,5,5))
for(phename in levels(factor(noiqs_probSibs2_1$Phenotype))){
  dat <- noiqs_probSibs2_1[noiqs_probSibs2_1$Phenotype == phename,]
  phename1 <- gsub("_sum", "", phename)
  manhattan(dat, main = phename1, ylim = c(5, 10), col = c("blue4", "orange3"), suggestiveline = F, genomewideline = -log10(5e-08), cex = 3, annotatePval = 5e-8, annotateTop = F, cex.axis = 0.8, cex.main = 3, cex.lab =2, cex.axis = 2
  )
}
dev.off()

png("figures/manhplot_probandsAndUnaffSibs_noIQ_1e-5_secondary_withPCs.png",width=18, height = 6, units = "in", res = 300)
par(mfrow=c(1,3))
par(mar=c(5,5,5,5))
for(phename in levels(factor(noiqs_probSibs2_2$Phenotype))){
  dat <- noiqs_probSibs2_2[noiqs_probSibs2_2$Phenotype == phename,]
  phename1 <- gsub("_sum", "", phename)
  manhattan(dat, main = phename1, ylim = c(5, 10), col = c("blue4", "orange3"), suggestiveline = F, genomewideline = -log10(5e-08), cex = 3, annotatePval = 5e-8, annotateTop = F, cex.axis = 0.8, cex.main = 3, cex.lab =2, cex.axis = 2
  )
}
dev.off()
```

### Association Summary {.tabset .tabset-fade .tabset-pills}
#### Fitting FSIQ
```{r}
datatable(iqs_probSibs2, rownames = FALSE, filter="top", options = list(pageLength = 5, scrollX=T) )
```

***
  
<br>

#### Not fitting FSIQ
```{r}
datatable(noiqs_probSibs2, rownames = FALSE, filter="top", options = list(pageLength = 5, scrollX=T) )
```

***
  
<br>

### Manhattan Plot {.tabset .tabset-fade .tabset-pills}
#### Primary Variable  {.tabset .tabset-fade .tabset-pills}
##### Fitting FSIQ
```{r out.width="100%", out.height="100%"}
grid.raster(readPNG("figures/manhplot_probandsAndUnaffSibs_adjIQ_1e-5_primary_withPCs.png"))
```

***
  
<br>

##### Not fitting FSIQ
```{r out.width="100%", out.height="100%"}
grid.raster(readPNG("figures/manhplot_probandsAndUnaffSibs_noIQ_1e-5_primary_withPCs.png"))
```


***
  
<br>

#### Secondary Variable  {.tabset .tabset-fade .tabset-pills}

##### Fitting FSIQ
```{r out.width="100%", out.height="100%"}
grid.raster(readPNG("figures/manhplot_probandsAndUnaffSibs_adjIQ_1e-5_secondary_withPCs.png"))
```

***
  
<br>

##### Not fitting FSIQ
```{r out.width="100%", out.height="100%"}
grid.raster(readPNG("figures/manhplot_probandsAndUnaffSibs_noIQ_1e-5_secondary_withPCs.png"))
```

***
  
<br>


## Europeans Only  {.tabset .tabset-fade .tabset-pills}
- GWAS was run on European individuals with N=3544.
- Signals with association p-value < 1e-5 will be shown.

```{r merge_GWAS_probandsSibs_EUR, eval=FALSE, include=FALSE}
setwd("/scratch/90days/uqywan67/auti_proj/SSC/outputs")

summs <- read.table("phenotype_variable_summary.txt", header = T, stringsAsFactors = F)

iqs_probSibs_EUR <- read.table("summary_probandsAndunaffSibs_p1e5_EUR.txt", stringsAsFactors = F)
noiqs_probSibs_EUR <- read.table("summary_probandsAndunaffSibs_p1e5_noIQ_EUR.txt", stringsAsFactors = F)
names(iqs_probSibs_EUR) <- names(noiqs_probSibs_EUR) <- c("Chr", "SNP", "bp", "A1", "A2", "Freq", "b", "se", "p", "Strategy", "Phenotype")

iqs_probSibs_EUR$Interests <- "Primary"
iqs_probSibs_EUR[iqs_probSibs_EUR$Phenotype %in% summs$Primary_Variable, "Interests"] <- "Primary"
iqs_probSibs_EUR[iqs_probSibs_EUR$Phenotype %in% summs$Secondary_Variables, "Interests"] <- "Secondary"

iqs_probSibs_EUR1 <- iqs_probSibs_EUR[grep("sum", iqs_probSibs_EUR$Phenotype),]
iqs_probSibs_EUR2 <- iqs_probSibs_EUR1[iqs_probSibs_EUR1$Strategy == "Strategy2",][,c(1:3, 6:12)]
names(iqs_probSibs_EUR2)[1:7] <- c("CHR", "SNP", "BP", "Freq",  "b", "se", "P")

write.table(iqs_probSibs_EUR2, file = "summary_probandsAndUnaffSibs_p1e5_strategy2_EUR.txt", col.names = T, row.names = F, quote = F, sep = "\t")
write.table(iqs_probSibs_EUR, file = "summary_probandsAndUnaffSibs_p1e5_EUR.txt", col.names = T, row.names = F, quote = F, sep = "\t")

noiqs_probSibs_EUR$Interests <- "Primary"
noiqs_probSibs_EUR[noiqs_probSibs_EUR$Phenotype %in% summs$Primary_Variable, "Interests"] <- "Primary"
noiqs_probSibs_EUR[noiqs_probSibs_EUR$Phenotype %in% summs$Secondary_Variables, "Interests"] <- "Secondary"


noiqs_probSibs_EUR1 <- noiqs_probSibs_EUR[grep("sum", noiqs_probSibs_EUR$Phenotype),]
noiqs_probSibs_EUR2 <- noiqs_probSibs_EUR1[noiqs_probSibs_EUR1$Strategy == "Strategy2",][,c(1:3, 6:12)]
names(noiqs_probSibs_EUR2)[1:7] <- c("CHR", "SNP", "BP", "Freq", "b", "se", "P")
#noiqs_probSibs_EUR2$zscore <- noiqs_probSibs_EUR2$b/noiqs_probSibs_EUR2$se
write.table(noiqs_probSibs_EUR2, file = "summary_probandsAndUnaffSibs_p1e5_noIQ_strategy2_EUR.txt", col.names = T, row.names = F, quote = F, sep = "\t")
write.table(noiqs_probSibs_EUR, file = "summary_probandsAndUnaffSibs_p1e5_noIQ_EUR.txt", col.names = T, row.names = F, quote = F, sep = "\t")

```


```{r include=FALSE}
iqs_probSibs_EUR2 <- read.table("../outputs/summary_probandsAndUnaffSibs_p1e5_strategy2_EUR.txt", header = T, stringsAsFactors = F)
iqs_probSibs_EUR2_1 <- iqs_probSibs_EUR2[iqs_probSibs_EUR2$Interests == "Primary",]
iqs_probSibs_EUR2_2 <- iqs_probSibs_EUR2[iqs_probSibs_EUR2$Interests == "Secondary",]

noiqs_probSibs_EUR2 <- read.table("../outputs/summary_probandsAndUnaffSibs_p1e5_noIQ_strategy2_EUR.txt", header = T, stringsAsFactors = F)
noiqs_probSibs_EUR2_1 <- noiqs_probSibs_EUR2[noiqs_probSibs_EUR2$Interests == "Primary",]
noiqs_probSibs_EUR2_2 <- noiqs_probSibs_EUR2[noiqs_probSibs_EUR2$Interests == "Secondary",]

```

```{r plot_manha_probSibs_EUR, eval=FALSE, include=FALSE}
##with IQ adjusted
png("figures/manhplot_probandsAndUnaffSibs_adjIQ_1e-5_primary_EUR_withPCs.png",width=12, height = 12, units = "in", res = 300)
par(mfrow=c(2,2))
par(mar=c(5,5,5,5))
for(phename in levels(factor(iqs_probSibs_EUR2_1$Phenotype))){
  dat <- iqs_probSibs_EUR2_1[iqs_probSibs_EUR2_1$Phenotype == phename,]
  phename1 <- gsub("_sum", "", phename)
  manhattan(dat, main = phename1, ylim = c(5, 10), col = c("blue4", "orange3"), suggestiveline = F, genomewideline = -log10(5e-08), cex = 3, annotatePval = 5e-8, annotateTop = F, cex.axis = 0.8, cex.main = 3, cex.lab =2, cex.axis = 2
  )
}
dev.off()

png("figures/manhplot_probandsAndUnaffSibs_adjIQ_1e-5_secondary_EUR_withPCs.png",width=18, height = 6, units = "in", res = 300)
par(mfrow=c(1,3))
par(mar=c(5,5,5,5))
for(phename in levels(factor(iqs_probSibs_EUR2_2$Phenotype))){
  dat <- iqs_probSibs_EUR2_2[iqs_probSibs_EUR2_2$Phenotype == phename,]
  phename1 <- gsub("_sum", "", phename)
  manhattan(dat, main = phename1, ylim = c(5, 10), col = c("blue4", "orange3"), suggestiveline = F, genomewideline = -log10(5e-08), cex = 3, annotatePval = 5e-8, annotateTop = F, cex.axis = 0.8, cex.main = 3, cex.lab =2, cex.axis = 2
  )
}
dev.off()

##without IQ adjusted
png("figures/manhplot_probandsAndUnaffSibs_noIQ_1e-5_primary_EUR_withPCs.png",width=12, height = 12, units = "in", res = 300)
par(mfrow=c(2,2))
par(mar=c(5,5,5,5))
for(phename in levels(factor(noiqs_probSibs_EUR2_1$Phenotype))){
  dat <- noiqs_probSibs_EUR2_1[noiqs_probSibs_EUR2_1$Phenotype == phename,]
  phename1 <- gsub("_sum", "", phename)
  manhattan(dat, main = phename1, ylim = c(5, 10), col = c("blue4", "orange3"), suggestiveline = F, genomewideline = -log10(5e-08), cex = 3, annotatePval = 5e-8, annotateTop = F, cex.axis = 0.8, cex.main = 3, cex.lab =2, cex.axis = 2
  )
}
dev.off()

png("figures/manhplot_probandsAndUnaffSibs_noIQ_1e-5_secondary_EUR_withPCs.png",width=18, height = 6, units = "in", res = 300)
par(mfrow=c(1,3))
par(mar=c(5,5,5,5))
for(phename in levels(factor(noiqs_probSibs_EUR2_2$Phenotype))){
  dat <- noiqs_probSibs_EUR2_2[noiqs_probSibs_EUR2_2$Phenotype == phename,]
  phename1 <- gsub("_sum", "", phename)
  manhattan(dat, main = phename1, ylim = c(5, 10), col = c("blue4", "orange3"), suggestiveline = F, genomewideline = -log10(5e-08), cex = 3, annotatePval = 5e-8, annotateTop = F, cex.axis = 0.8, cex.main = 3, cex.lab =2, cex.axis = 2
  )
}
dev.off()
```

### Association Summary {.tabset .tabset-fade .tabset-pills}
#### Fitting FSIQ
```{r}
datatable(iqs_probSibs_EUR2, rownames = FALSE, filter="top", options = list(pageLength = 5, scrollX=T) )
```

***
  
<br>

#### Not fitting FSIQ
```{r}
datatable(noiqs_probSibs_EUR2, rownames = FALSE, filter="top", options = list(pageLength = 5, scrollX=T) )
```

***
  
<br>

### Manhattan Plot {.tabset .tabset-fade .tabset-pills}
#### Primary Variable  {.tabset .tabset-fade .tabset-pills}
##### Fitting FSIQ
```{r out.width="100%", out.height="100%"}
grid.raster(readPNG("figures/manhplot_probandsAndUnaffSibs_adjIQ_1e-5_primary_EUR_withPCs.png"))
```

***
  
<br>

##### Not fitting FSIQ
```{r out.width="100%", out.height="100%"}
grid.raster(readPNG("figures/manhplot_probandsAndUnaffSibs_noIQ_1e-5_primary_EUR_withPCs.png"))
```


***
  
<br>

#### Secondary Variable  {.tabset .tabset-fade .tabset-pills}

##### Fitting FSIQ
```{r out.width="100%", out.height="100%"}
grid.raster(readPNG("figures/manhplot_probandsAndUnaffSibs_adjIQ_1e-5_secondary_EUR_withPCs.png"))
```

***
  
<br>

##### Not fitting FSIQ
```{r out.width="100%", out.height="100%"}
grid.raster(readPNG("figures/manhplot_probandsAndUnaffSibs_noIQ_1e-5_secondary_EUR_withPCs.png"))
```

***
  
<br>

## Heritability Estimation

***

<br>


