<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>SSC imputation pipeline</title>

<script src="site_libs/header-attrs-2.6/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/paper.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>





<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 64px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 69px;
  margin-top: -69px;
}
.section h2 {
  padding-top: 69px;
  margin-top: -69px;
}
.section h3 {
  padding-top: 69px;
  margin-top: -69px;
}
.section h4 {
  padding-top: 69px;
  margin-top: -69px;
}
.section h5 {
  padding-top: 69px;
  margin-top: -69px;
}
.section h6 {
  padding-top: 69px;
  margin-top: -69px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Austim Speaks | SSC</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="index.html">Introduction</a>
</li>
<li>
  <a href="SSC_QC_pipeline.html">Raw Genotype</a>
</li>
<li>
  <a href="impute_pipe_SSC.html">Imputation</a>
</li>
<li>
  <a href="Phenotype.html">Phenotype</a>
</li>
<li>
  <a href="GWAS.html">GWAS</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">SSC imputation pipeline</h1>

</div>


<hr />
<div id="data-structure" class="section level1" number="1">
<h1 number="1"><span class="header-section-number">1</span> Data structure</h1>
<ul>
<li>Individuals from 2591 SSC families were genotyped on three chips. Note that members of each family were analyzed on the same array.
<ul>
<li>Illumina 1Mv3 Duo (1189 families), 4626 people (2703 males, 1923 females), 1199033 SNPs</li>
<li>Illumina 1Mv1 (333 families), 1354 people (801 males, 553 females), 1072814 SNPs</li>
<li>Illumina HumanOmni2.5M (1069 families), 4240 people (2490 males, 1750 females), 2440283 SNPs</li>
<li>coded with ATGC, no need to recode</li>
<li>chromosome are from chr1 to chr26, of which autosome 1-22, X:23 &amp; 25 will be imputed (** –output-chr M can recode as X instead of 23**)</li>
<li><strong>SNP IDs</strong>: pay attention to Omini2.5, (–update-name from format as SNP(chr)-pos to kgp identifiers should be done first), later kgp identifiers should be converted to rsids.</li>
<li>All individuals are included for imputation</li>
</ul></li>
<li>Imputation will be done separately for each chip.</li>
</ul>
<hr />
<p><br></p>
</div>
<div id="pre-imputation" class="section level1" number="2">
<h1 number="2"><span class="header-section-number">2</span> Pre-imputation</h1>
<p><br></p>
<div id="strand-flip-liftover" class="section level2" number="2.1">
<h2 number="2.1"><span class="header-section-number">2.1</span> Strand flip &amp; liftover</h2>
<ul>
<li>From build 36 to build 37, from minus strand to plus strand)</li>
<li>SSC was assembled based on build 36 (confirmed with <a href="https://www.ncbi.nlm.nih.gov/snp/?cmd=search">dbSNP</a>)</li>
<li><a href="https://www.well.ox.ac.uk/~wrayner/strand/">strand files</a>
<ul>
<li>Human1Mv1_C-b37.strand is used for both 1Mv1 and 1Mv3</li>
<li>HumanOmni2.5-8v1_A-b37.Source.strand is used for Omni2.5</li>
</ul></li>
<li>Check the output files from <a href="https://www.well.ox.ac.uk/~wrayner/strand/update_build.sh">update_build.sh</a>
<ul>
<li>remove chr 24</li>
<li>recode chr 23 to chr X</li>
<li>update-name for kgp identifiers in Omni2.5 (when there are a multitude of SNPs coded as kgp)</li>
</ul></li>
</ul>
<hr />
<p><br></p>
</div>
<div id="genotype-qc" class="section level2 tabset tabset-fade tabset-pills" number="2.2">
<h2 class="tabset tabset-fade tabset-pills" number="2.2"><span class="header-section-number">2.2</span> Genotype QC</h2>
<ul>
<li>Filters: --geno 0.05 --hwe 1e-6 --mind 0.1 --maf 0.01.</li>
<li>Then recode to .vcf files (separate autosome and X-chromosome for now)</li>
</ul>
<div id="illumina-1mv3" class="section level3" number="2.2.1">
<h3 number="2.2.1"><span class="header-section-number">2.2.1</span> Illumina 1Mv3</h3>
<hr />
<ul>
<li>0 people removed due to missing genotype data (--mind).</li>
<li>Total genotyping rate is 0.990957.</li>
<li>15033 variants removed due to missing genotype data (--geno).</li>
<li>126158 variants removed due to minor allele threshold(s).</li>
<li>--hwe: 38392 variants removed due to Hardy-Weinberg exact test.</li>
<li><strong>859161 variants and 4626 people pass filters and QC.</strong></li>
</ul>
<hr />
<p><br></p>
</div>
<div id="illumina-1mv1" class="section level3" number="2.2.2">
<h3 number="2.2.2"><span class="header-section-number">2.2.2</span> Illumina 1Mv1</h3>
<hr />
<ul>
<li>0 people removed due to missing genotype data (--mind).</li>
<li>Total genotyping rate is 0.976306.</li>
<li>29018 variants removed due to missing genotype data (--geno).</li>
<li>136291 variants removed due to minor allele threshold(s).</li>
<li>--hwe: 6476 variants removed due to Hardy-Weinberg exact test.</li>
<li><strong>898063 variants and 1354 people pass filters and QC.</strong></li>
</ul>
<hr />
<p><br></p>
</div>
<div id="illumina-omni2.5m" class="section level3" number="2.2.3">
<h3 number="2.2.3"><span class="header-section-number">2.2.3</span> Illumina Omni2.5M</h3>
<hr />
<ul>
<li>0 people removed due to missing genotype data (--mind).</li>
<li>Total genotyping rate is 0.99696.</li>
<li>7885 variants removed due to missing genotype data (--geno).</li>
<li>723923 variants removed due to minor allele threshold(s).</li>
<li>--hwe: 53449 variants removed due to Hardy-Weinberg exact test.</li>
<li><strong>1575751 variants and 4240 people pass filters and QC.</strong></li>
</ul>
<hr />
<p><br></p>
</div>
</div>
<div id="check-and-fix-the-ref-allele" class="section level2" number="2.3">
<h2 number="2.3"><span class="header-section-number">2.3</span> Check and fix the REF allele</h2>
<ul>
<li>only for sanger imputation server</li>
<li>using bcftools</li>
<li>check with <a href="ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/technical/reference/human_g1k_v37.fasta.gz">HRC build37</a></li>
</ul>
<style>
div.blue { background-color:#e6f0ff; border-radius: 5px; padding: 20px;}
</style>
<div class="blue">
<p><strong>The whole pre-Imputation QC process stored in /gpfs1/scratch/group30days/cnsg_park/uqywan/auti/scripts/ssc_preImp_QC.R</strong></p>
</div>
<hr />
<p><br></p>
</div>
<div id="submit-for-imputation" class="section level2" number="2.4">
<h2 number="2.4"><span class="header-section-number">2.4</span> Submit for imputation</h2>
<ul>
<li><a href="https://imputation.sanger.ac.uk">Sanger imputation server</a></li>
<li>Using Globus to transfer files (##sanger to ##flash)</li>
<li>HRC reference &amp; Eagle v2.4 for phasing</li>
</ul>
<hr />
<p><br></p>
</div>
</div>
<div id="after-imputation" class="section level1" number="3">
<h1 number="3"><span class="header-section-number">3</span> After Imputation</h1>
<ul>
<li>recode .vcf to plink format</li>
<li>extract info scores in .info</li>
<li>fill missing SNP IDs in .bim/.info</li>
<li>update .fam file in case there are two "_" in the FID_IID</li>
<li>QC for each chip (filters: --info 0.03)
<ul>
<li>1Mv1: 29,613,517 SNPs left</li>
<li>1Mv3: 31,605,064 SNPs left</li>
<li>Omni2.5: 30,466,266 SNPs left</li>
<li>23,460,315 SNPs overlapped</li>
</ul></li>
<li>merge .bfile of three chips
<ul>
<li>remaining 22,274,458 SNPs after QC for --geno 0.05 --mac 1 --hwe 1e-6</li>
<li>remaining 7,480,061 SNPs after QC for --geno 0.05 --maf 0.01 --hwe 1e-6</li>
</ul></li>
</ul>
<hr />
<p><br></p>
<div id="frequency-distrubtion" class="section level2" number="3.1">
<h2 number="3.1"><span class="header-section-number">3.1</span> Frequency distrubtion</h2>
<ul>
<li>overlapped SNPs between SSC_imputed and HRC (~7.2M SNPs with MAF &gt; 0.01)</li>
<li>based on same allele</li>
<li><strong>0 SNPs with MAF difference &gt; 0.2</strong></li>
</ul>
<p><img src="figures/frqCheck.png" width="80%" height="60%" /></p>
<hr />
<p><br></p>
</div>
<div id="pca" class="section level2 tabset tabset-fade tabset-pills" number="3.2">
<h2 class="tabset tabset-fade tabset-pills" number="3.2"><span class="header-section-number">3.2</span> PCA</h2>
<ul>
<li>Project the first 3 PCs based on pruned HapMap3 SNPs onto 1000G</li>
<li>Using K-means to calculate distance</li>
<li>Assign ancestry based on posterior probability 0.9</li>
</ul>
<div id="all-individuals" class="section level3" number="3.2.1">
<h3 number="3.2.1"><span class="header-section-number">3.2.1</span> All individuals</h3>
<pre class="r"><code>toAssign &lt;- read.table(&quot;../outputs/ssc_anc.txt&quot;, header = T, stringsAsFactors = F)

#pca_all &lt;- 
ggplot(data = toAssign, aes(x = PC1, y = PC2, color = Assigned, shape = Group)) +
  geom_point() +
  ylab(&quot;PC2&quot;) +
  xlab(&quot;PC1&quot;)</code></pre>
<p><img src="impute_pipe_SSC_files/figure-html/pca_all-1.png" width="80%" height="60%" /></p>
<hr />
<p><br></p>
</div>
<div id="probands-only" class="section level3 tabset tabset-fade tabset-pills" number="3.2.2">
<h3 class="tabset tabset-fade tabset-pills" number="3.2.2"><span class="header-section-number">3.2.2</span> Probands only</h3>
<div id="self-reported-vs-ancestry-assigned" class="section level4" number="3.2.2.1">
<h4 number="3.2.2.1"><span class="header-section-number">3.2.2.1</span> Self Reported VS Ancestry Assigned</h4>
<pre class="r"><code>toAssign1 &lt;- read.table(&quot;../outputs/proband_anc.txt&quot;, header = T, stringsAsFactors = F)
kable(table(toAssign1$Self_Reported, toAssign1$Assigned))</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">AFR</th>
<th align="right">AMR</th>
<th align="right">EAS</th>
<th align="right">EUR</th>
<th align="right">SAS</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">african-amer</td>
<td align="right">91</td>
<td align="right">7</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">asian</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">60</td>
<td align="right">0</td>
<td align="right">47</td>
</tr>
<tr class="odd">
<td align="left">more-than-one-race</td>
<td align="right">11</td>
<td align="right">100</td>
<td align="right">2</td>
<td align="right">44</td>
<td align="right">45</td>
</tr>
<tr class="even">
<td align="left">native-american</td>
<td align="right">0</td>
<td align="right">4</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">native-hawaiian</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">not-specified</td>
<td align="right">1</td>
<td align="right">16</td>
<td align="right">0</td>
<td align="right">2</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">other</td>
<td align="right">4</td>
<td align="right">91</td>
<td align="right">0</td>
<td align="right">14</td>
<td align="right">12</td>
</tr>
<tr class="even">
<td align="left">white</td>
<td align="right">1</td>
<td align="right">116</td>
<td align="right">0</td>
<td align="right">1907</td>
<td align="right">8</td>
</tr>
</tbody>
</table>
<hr />
<p><br></p>
</div>
<div id="pca-plot-based-on-assigned-ancestry" class="section level4" number="3.2.2.2">
<h4 number="3.2.2.2"><span class="header-section-number">3.2.2.2</span> PCA plot based on assigned ancestry</h4>
<pre class="r"><code>#pca_prob &lt;- 
ggplot(data = toAssign1, aes(x = PC1, y = PC2, color = Assigned)) +
  geom_point() +
  ylab(&quot;PC2&quot;) +
  xlab(&quot;PC1&quot;)</code></pre>
<p><img src="impute_pipe_SSC_files/figure-html/pca_prob-1.png" width="80%" height="60%" /></p>
<hr />
<p><br></p>
</div>
</div>
</div>
<div id="relatedness-in-probands" class="section level2" number="3.3">
<h2 number="3.3"><span class="header-section-number">3.3</span> Relatedness in Probands</h2>
<ul>
<li>Generate GRM based on LD pruned SNPs</li>
<li>2206 probands with pairwise GRM &lt; 0.05</li>
</ul>
<p><img src="figures/SSC_proband_prune.3.png" width="80%" height="60%" /></p>
<hr />
<style>
div.blue { background-color:#e6f0ff; border-radius: 5px; padding: 20px;}
</style>
<div class="blue">
<p>**The whole process stored in /gpfs1/scratch/group30days/cnsg_park/uqywan/auti/scripts/afterImp_QC*.R**</p>
</div>
<hr />
<p><br></p>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
